### PR TITLE
Refactor imds/gcp to no longer depend on lib/cloud/gcp

### PR DESCRIPTION
**Pull request by rosstimothy**
The imds pacakges are inteneded to be very lightweight and consumable by client tools without any additional cloud specific dependencies. By inverting dependencies such that lib/cloud/gcp depends on lib/cloud/imds/gcp client tools no longer transatively import the very large `cloud.google.com/go/compute/apiv1`.

The depguard rules were also updated to prevent any accidental regressions.

### CODE REVIEW

File: lib/cloud/gcp/vm.go [hunk 1]
```diff
@@ -452,10 +396,22 @@ func (clt *instancesClient) AddSSHKey(ctx context.Context, req *SSHKeyRequest) e
...
+	items := make([]*computepb.Items, 0, len(req.Instance.Items))
+	for k, v := range req.Instance.Items {
+		items = append(items, &computepb.Items{Key: googleapi.String(k), Value: googleapi.String(v)})
+	}
+
+	metadata := &computepb.Metadata{
+		Fingerprint: googleapi.String(req.Instance.FingerPrint),
+		Items:       items,
+		Kind:        googleapi.String(req.Instance.Kind),
+	}
```
Comment (atburke) [inline lib/cloud/gcp/vm.go:410]: This section repeats in `RemoveSSHKey` and can probably be moved into a helper function.

File: lib/cloud/imds/gcp/imds.go [hunk 2]
```diff
@@ -53,24 +52,75 @@ func (rt contextRoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
...
+	ServiceAccount string
+	// Labels is the instance's labels.
+	Labels map[string]string
+
+	InternalIPAddress string
+	ExternalIPAddress string
+	HostKeys          []ssh.PublicKey
+	FingerPrint       string
+	Kind              string
+	Items             map[string]string
```
Comment (atburke) [inline lib/cloud/imds/gcp/imds.go:101]: Nit: godoc

Comment (atburke) [inline lib/cloud/imds/gcp/imds.go:101]:
```suggestion
	MetadataItems     map[string]string
```
or something like that, so it's clearer what "items" we're talking about.

File: lib/cloud/imds/gcp/imds.go [hunk 3]
```diff
@@ -53,24 +52,75 @@ func (rt contextRoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
...
+	ProjectID string
+	// ServiceAccount is the email address of the VM's service account, if any.
+	ServiceAccount string
+	// Labels is the instance's labels.
+	Labels map[string]string
+
+	InternalIPAddress string
+	ExternalIPAddress string
+	HostKeys          []ssh.PublicKey
+	FingerPrint       string
```
Comment (atburke) [inline lib/cloud/imds/gcp/imds.go:99]: Nit:
```suggestion
	Fingerprint       string
```

File: lib/cloud/imds/gcp/imds.go [hunk 4]
```diff
@@ -121,23 +171,20 @@ func (client *InstanceMetadataClient) GetTags(ctx context.Context) (map[string]s
...
 	// Get labels.
 	var gcpLabels map[string]string
-	inst, err := client.instancesClient.GetInstance(ctx, req)
+	inst, err := client.instanceGetter.GetInstance(ctx, req)
 	if err == nil {
 		gcpLabels = inst.Labels
 	} else if trace.IsAccessDenied(err) {
-		client.labelPermissionErrorOnce.Do(func() {
-			slog.WarnContext(ctx, "Access denied to instance labels, does the instance have compute.instances.get permission?")
-		})
```
Comment (atburke) [inline lib/cloud/imds/gcp/imds.go:130]: Getting rid of the once here means that we'll get this message every time the node tries to update its labels. Just wanted to make sure that's the effect you intended.

Comment (atburke) [reply lib/cloud/imds/gcp/imds.go:130]: You're not missing anything. It sounds like we're good then.

**Review by atburke — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**